`py install --update` fails to update when no specific version is given

I installed an older version of 3.12 to see what happens when I try to update everything and got this error:

`[ERROR] An error occurred. Please check any output above, or the log file, and try again.`

Here's the log:

```
>> Writing logs to C:\Users\ducks\AppData\Local\Temp\python_install_20250418093606_12820.log
>> Python installation manager 0.1a16
Copyright (c) Python Software Foundation. All Rights Reserved.

BEGIN install_command.execute: []
Reading installs from C:\Users\ducks\AppData\Local\Python
Found 1 install
Calculating default install
default_platform '-64' matched 1 install
Default install will be pythoncore-3.12-64
>> No tags provided, updating all installs:
Searching for Python with ID pythoncore-3.12-64
Fetching: file:///C:/Program Files/WindowsApps/PythonSoftwareFoundation.PythonManager_0.1.0.176_x64__7qnp2ymwnadz2/./bundled/index.json
!! An error occurred. Please check any output above, or the log file, and try again.
>  Log file for this session: C:\Users\ducks\AppData\Local\Temp\python_install_20250418093606_12820.log
>  If you cannot resolve it yourself, please report the error with your log file at https://discuss.python.org/t/77900/ (during testing).
TRACEBACK:
Traceback (most recent call last):
  File "manage\install_command.py", line 624, in execute
TypeError: _install_one() missing 1 required positional argument: 'install'

END install_command.execute
SILENCED ERROR
Traceback (most recent call last):
  File "manage\install_command.py", line 624, in execute
TypeError: _install_one() missing 1 required positional argument: 'install'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "manage\__init__.py", line 42, in main
  File "manage\commands.py", line 732, in execute
  File "manage\install_command.py", line 673, in execute
  File "manage\install_command.py", line 491, in _fatal_install_error
SystemExit: 1
```

`py install --update 3.12` works as expected.

--- a/_msbuild.py
+++ b/_msbuild.py
@@ -180,18 +180,12 @@ PACKAGE = Package('python-manager',
 )
 
 
-DLL_NAMES = {
-    "cp313": "python313",
-    "cp314": "python314",
-}
+DLL_NAME = "python314"
+EMBED_URL = "https://www.python.org/ftp/python/3.14.0/python-3.14.0a7-embed-amd64.zip"
 
-
-EMBED_URLS = {
-    "cp313-cp313-win_amd64": "https://www.python.org/ftp/python/3.13.2/python-3.13.2-embed-amd64.zip",
-    "cp313-cp313-win_arm64": "https://www.python.org/ftp/python/3.13.2/python-3.13.2-embed-arm64.zip",
-    "cp314-cp314-win_amd64": "https://www.python.org/ftp/python/3.14.0/python-3.14.0a7-embed-amd64.zip",
-    "cp314-cp314-win_arm64": "https://www.python.org/ftp/python/3.14.0/python-3.14.0a7-embed-arm64.zip",
-}
+def can_embed(tag):
+    "Return False if tag doesn't match DLL_NAME and EMBED_URL."
+    return tag == "cp314-cp314-win_amd64"
 
 
 def get_commands():
@@ -333,14 +327,17 @@ def init_PACKAGE(tag=None):
     update_file(cmds_h, cmds_txt)
 
     # BUNDLE EMBEDDABLE DISTRO
-    dll_name = DLL_NAMES[tag.partition("-")[0]]
-    PACKAGE.find("py-manager/ItemDefinition(Link)").options["DelayLoadDLLs"] = f"{dll_name}.dll"
-    PACKAGE.find("pyw-manager/ItemDefinition(Link)").options["DelayLoadDLLs"] = f"{dll_name}.dll"
+    if not can_embed(tag):
+        print("[WARNING] Unable to bundle embeddable distro for this runtime.")
+        return
+
+    PACKAGE.find("py-manager/ItemDefinition(Link)").options["DelayLoadDLLs"] = f"{DLL_NAME}.dll"
+    PACKAGE.find("pyw-manager/ItemDefinition(Link)").options["DelayLoadDLLs"] = f"{DLL_NAME}.dll"
 
     embed_files = [tmpdir / tag / n for n in [
-        f"{dll_name}.dll",
-        f"{dll_name}.zip",
-        f"{dll_name}._pth",
+        f"{DLL_NAME}.dll",
+        f"{DLL_NAME}.zip",
+        f"{DLL_NAME}._pth",
     ]]
     runtime_files = [tmpdir / tag / n for n in [
         "vcruntime140.dll",
@@ -351,7 +348,7 @@ def init_PACKAGE(tag=None):
         from zipfile import ZipFile
         package = tmpdir / tag / "package.zip"
         package.parent.mkdir(exist_ok=True, parents=True)
-        urlretrieve(EMBED_URLS[tag], package)
+        urlretrieve(EMBED_URL, package)
         with ZipFile(package) as zf:
             for f in [*embed_files, *runtime_files]:
                 f.write_bytes(zf.read(f.name))

--- a/src/manage/install_command.py
+++ b/src/manage/install_command.py
@@ -594,7 +594,8 @@ def execute(cmd):
                 if cmd.repair:
                     LOGGER.verbose("No tags provided, repairing all installs:")
                     for install in installed:
-                        _install_one(cmd, install)
+                        # Only try to redownload from the same source
+                        _install_one(cmd, install.get('source'), install)
                     # Fallthrough is safe - cmd.tags is empty
                 elif cmd.update:
                     LOGGER.verbose("No tags provided, updating all installs:")
@@ -621,7 +622,7 @@ def execute(cmd):
                             raise RuntimeError("All install sources failed, nothing can be updated.")
                         if update:
                             if update['sort-version'] > install['sort-version']:
-                                _install_one(cmd, update)
+                                _install_one(cmd, source, update)
                             else:
                                 LOGGER.verbose("%s is already up to date.", install['display-name'])
                         else:

